scripts/generate-zod-schemas.js

```diff
@@ -0,0 +1,252 @@
+      ? `  /** ${prop.description} */\n  `
+      : '  ';
+    
+    return `${comment}${propName}: ${zodType}`;
+  });
+
+  const schemaDefinition = `export const ${metadata.name}Schema = z.object({\n${propSchemas.join(',\n')}\n})`;
+  const typeDefinition = `export type ${metadata.name}Props = z.infer<typeof ${metadata.name}Schema>`;
+  
+  return `${schemaDefinition}\n\n${typeDefinition}`;
+}
+
+// Create individual component directory with Zod schema
+function createComponentZodSchema(componentName, componentData) {
+  const componentDir = join(projectRoot, 'components', componentName);
+  
+  // Create directory if it doesn't exist
+  if (!existsSync(componentDir)) {
+    mkdirSync(componentDir, { recursive: true });
+  }
+
+  // Generate Zod Schema
+  const zodSchema = convertMetadataToZodSchema(componentData);
+  
+  // Write schema.zod.ts
+  const schemaPath = join(componentDir, 'schema.zod.ts');
+  const content = `// Auto-generated Zod schema for ${componentName}
+// Generated on: ${new Date().toISOString()}
+import { z } from 'zod'
+
+${zodSchema}
+`;
+  
+  writeFileSync(schemaPath, content);
+  
+  return { componentName, propsCount: (componentData.props || []).length };
+}
+
+// Generate main zod/index.ts that exports all component schemas
+function generateZodIndex(componentNames) {
+  const indexContent = `// Auto-generated Zod schemas index for PatternFly components
+// Generated on: ${new Date().toISOString()}
+import { z } from 'zod'
+
+${componentNames.map(name => 
+  `export * from '../components/${name}/schema.zod.js'`
+).join('\n')}
+
+// Export all component names for convenience
+export const componentNames = [
+${componentNames.map(name => `  '${name}'`).join(',\n')}
+] as const
+
+// Helper to get schema by component name
+export function getComponentSchema(componentName: string) {
+  switch (componentName) {
+${componentNames.map(name => 
+    `    case '${name}': return import('../components/${name}/schema.zod.js').then(m => m.${name}Schema);`
+  ).join('\n')}
+    default: throw new Error(\`Unknown component: \${componentName}\`);
+  }
+}
+`;
+
+  // Create zod directory
+  const zodDir = join(projectRoot, 'zod');
+  if (!existsSync(zodDir)) {
+    mkdirSync(zodDir, { recursive: true });
+  }
+
+  const zodIndexPath = join(zodDir, 'index.ts');
+  writeFileSync(zodIndexPath, indexContent);
+}
+
+// Main conversion function
+function convertAllComponentsToZod(inputFile) {
+  console.log('📖 Reading metadata file for Zod schema generation...');
+  
+  if (!existsSync(inputFile)) {
+    console.error(`❌ Input file not found: ${inputFile}`);
+    console.log('💡 Please provide a JSON file with component metadata');
+    process.exit(1);
+  }
+
+  const metadata = JSON.parse(readFileSync(inputFile, 'utf8'));
+  
+  console.log(`🔍 Found ${Object.keys(metadata).length} components`);
+  
+  const processedComponents = [];
+  let convertedCount = 0;
+
+  Object.entries(metadata).forEach(([componentName, componentData]) => {
```

Comment:
Also a bit of a nit, but my gut says that this could be done more cleanly using a map than using a forEach and mutating variables outside of the loop.

Reply:
Thank you @wise-king-sullyman - switched to a `reduce` with internal encapsulation.